gmso/formats/lammpsdata.py

```diff
@@ -1,33 +1,235 @@
 """Read and write LAMMPS data files."""
 from __future__ import division
 
+import copy
 import datetime
+import re
 import warnings
+from pathlib import Path
 
 import numpy as np
 import unyt as u
-from sympy import simplify, sympify
+from sympy import Symbol
+from unyt import UnitRegistry
 from unyt.array import allclose_units
 
+import gmso
+from gmso.abc.abstract_site import MoleculeType
 from gmso.core.angle import Angle
 from gmso.core.angle_type import AngleType
 from gmso.core.atom import Atom
 from gmso.core.atom_type import AtomType
 from gmso.core.bond import Bond
 from gmso.core.bond_type import BondType
 from gmso.core.box import Box
+from gmso.core.dihedral import Dihedral
 from gmso.core.element import element_by_mass
+from gmso.core.improper import Improper
 from gmso.core.topology import Topology
+from gmso.core.views import PotentialFilters
+
+pfilter = PotentialFilters.UNIQUE_SORTED_NAMES
+from gmso.exceptions import NotYetImplementedWarning
 from gmso.formats.formats_registry import loads_as, saves_as
 from gmso.lib.potential_templates import PotentialTemplateLibrary
+from gmso.utils.compatibility import check_compatibility
 from gmso.utils.conversions import (
     convert_opls_to_ryckaert,
     convert_ryckaert_to_opls,
 )
+from gmso.utils.decorators import mark_WIP
+
+reg = UnitRegistry()
+dim = u.dimensions.current_mks * u.dimensions.time
+conversion = 1 * getattr(u.physical_constants, "elementary_charge").value
+reg.add(
+    "elementary_charge",
+    base_value=conversion,
+    dimensions=dim,
+    tex_repr=r"\rm{e}",
+)
+conversion = 1 * getattr(u.physical_constants, "boltzmann_constant_mks").value
+dim = u.dimensions.energy / u.dimensions.temperature
+reg.add(
+    "kb", base_value=conversion, dimensions=dim, tex_repr=r"\rm{kb}"
+)  # boltzmann temperature
+conversion = (
+    4
+    * np.pi
+    * getattr(u.physical_constants, "reduced_planck_constant").value ** 2
+    * getattr(u.physical_constants, "eps_0").value
+    / (
+        getattr(u.physical_constants, "electron_charge").value ** 2
+        * getattr(u.physical_constants, "electron_mass").value
+    )
+)
+dim = u.dimensions.length
+reg.add(
+    "a0", base_value=conversion, dimensions=dim, tex_repr=r"\rm{a0}"
+)  # bohr radius
+conversion = (
+    getattr(u.physical_constants, "reduced_planck_constant").value ** 2
+    / u.Unit("a0", registry=reg).base_value ** 2
+    / getattr(u.physical_constants, "electron_mass").value
+)
+dim = u.dimensions.energy
+reg.add(
+    "Ehartree", base_value=conversion, dimensions=dim, tex_repr=r"\rm{Ehartree}"
+)  # Hartree energy
+conversion = np.sqrt(
+    10**9 / (4 * np.pi * getattr(u.physical_constants, "eps_0").value)
+)
+dim = u.dimensions.charge
+reg.add(
+    "Statcoulomb_charge",
+    base_value=conversion,
+    dimensions=dim,
+    tex_repr=r"\rm{Statcoulomb_charge}",
+)  # Static charge
```

Comment:
Upon chatting with @daico007,  I think all of this is going to get moved to the units module in an upcoming PR. So, I'll leave this here for now, as it's not worth the effort to move this right before it will all get moved again.